### PR TITLE
:bug: gateway refactor default oauth2 config

### LINKED ISSUE #193: [Bug] Gateway not starting with Profile no security
### Bug description

SSO_ISSUER_URL  not set the Gateway is not starting. Even if I want only no_security

### Expected behaviour

start gateway or set a default value

### Steps for reproduction

siehe oben

### Affected version

1.0.0

### Affected component

refarch-gateway

### Last working version (optional)

_No response_

### Operating system

_No response_

### Browser

_No response_

### Relevant log output (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional context (optional)

_No response_

### No duplicate

- [X] I confirm that this issue is not a duplicate

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Refactor gateway default oauth2 config to fix `no-security` profile.

**Reference**

Issues closes #193

### CODE REVIEW

File: refarch-gateway/README.md [hunk 1]
```diff
@@ -41,17 +41,40 @@ Beside the default behaviour there are some special route prefixes which are han
...
+| `INFO_APPSWITCHER_URL` (optional)                         | App switcher url for usage in refarch frontend.    | `https://appswitcher.muenchen.de`                                       |
+
+### Security
+
+For authentication via SSO, OAuth2 needs to be configured.
+See following example or the [according Spring documentation](https://docs.spring.io/spring-security/reference/servlet/oauth2/index.html).
+
+```yaml
+spring:
+  # optional with profile "no-security"
```
Comment (devtobi) [inline refarch-gateway/README.md:63]: This comment suggests that the configuration is used when `no-security` is active, but actually quite the opposite is the case, right? Stuff like issuer URI and SSO provider information is used when used with security,, so I'd rewrite the comment

File: refarch-gateway/README.md [hunk 2]
```diff
@@ -41,17 +41,40 @@ Beside the default behaviour there are some special route prefixes which are han
...
+  security:
+    oauth2:
+      resourceserver:
+        jwt:
+          issuer-uri: "https://sso.muenchen.de/auth/realms/muenchen.de"
```
Comment (devtobi) [inline refarch-gateway/README.md:68]: Should be maybe provide a more general domain here? like `sso.domain.com/auth/...`?

File: refarch-gateway/README.md [hunk 3]
```diff
@@ -41,17 +41,40 @@ Beside the default behaviour there are some special route prefixes which are han
...
+| `SPRING_CLOUD_GATEWAY_ROUTES_<index>_PREDICATES_<index>`  | Route predicates i.e. matcher.                     | `Path=/api/backend-service/**`                                          |
+| `SPRING_CLOUD_GATEWAY_ROUTES_<index>_FILTERS_<index>`     | List of filters applied to the route.              | `RewritePath=/api/backend-service/(?<urlsegments>.*), /$\{urlsegments}` |
+| `ALLOWED_ORIGINS_PUBLIC` (optional)                       | List of urls allowed as origin for public routes.  | `https://*.muenchen.de,http://localhost:*`                              |
+| `ALLOWED_ORIGINS_CLIENTS` (optional)                      | List of urls allowed as origin for clients routes. | `https://*.muenchen.de,http://localhost:*`                              |
+| `REFARCH_SECURITY_CSRFWHITELISTED_<index>` (optional)     | List of routes to disable csrf protection for.     | `/example/**`                                                           |
+| `INFO_APPSWITCHER_URL` (optional)                         | App switcher url for usage in refarch frontend.    | `https://appswitcher.muenchen.de`                                       |
```
Comment (devtobi) [inline refarch-gateway/README.md:54]: Maybe provide a more general domain here as well. By the way, why is the AppSwitcher URL needed for the API gateway? Isn't the frontend directly requesting the AppSwicher-Server instance and not the API gateway?

Comment (simonhir) [reply refarch-gateway/README.md:54]: I don't know as i never used the appswitcher. This came from the old template.

Comment (devtobi) [reply refarch-gateway/README.md:54]: Okay we should investigate later on and maybe create a follow up issue to remove the AppSwitcher stuff if possible. 😄

Comment (simonhir) [reply refarch-gateway/README.md]: See #203

File: refarch-gateway/README.md (hunk 1 above)
Comment (simonhir) [reply refarch-gateway/README.md]: Refactored

File: refarch-gateway/README.md (hunk 2 above)
Comment (simonhir) [reply refarch-gateway/README.md]: Done

**Review by devtobi — APPROVED** [review]
LGTM

**Pull request merged**